src/quotes_handler.rs

```diff
@@ -183,20 +228,34 @@ pub async fn integrity(
             data.clone(),
         )?;
 
-        let mut quote = KeylimeIntegrityQuote::from_id_quote(
-            quote,
-            read_to_string(IMA_ML)?,
-        );
-
-        quote.pubkey = String::from_utf8(
-            data.pub_key
-                .public_key_to_pem()
+        // pubkey should only be sent for the first quote request; otherwise
+        // the verifier will keep sending a v key every time. we check whether
+        // it's the first quote request by seeing if the symmetric key has been
+        // derived (which happens after both u and v key are received once).
+        let symm_key = data.payload_symm_key.lock().unwrap(); //#[allow_ci]
+        if *symm_key == [0u8; KEY_LEN] {
+            let quote = KeylimeIntegrityQuotePreAttestation::from_id_quote(
+                quote,
+                read_to_string(IMA_ML)?,
+                String::from_utf8(
+                    data.pub_key
+                        .public_key_to_pem()
+                        .map_err(KeylimeError::from)?,
+                )
                 .map_err(KeylimeError::from)?,
-        )
-        .map_err(KeylimeError::from)?;
+            );
+            let response = JsonIntegWrapperPreAttestation::new(quote);
+            info!("GET integrity quote returning 200 response");
```

Comment:
If I'm reading this correctly we will _always_ get a response of 200 since it's hard coded in the `JsonIntegWrapperPreAttestation`, correct? If that's true, why log on it? Same question on the `JsonIntegWrapperPostAttestation` log line later.

Reply:
@ashcrow From my understanding and testing of the handler, if it encounters an error, it will return an HTTP code 500 Internal Server Error with the error message as part of the returned object. If it reaches this line, all the processing has succeeded, so we return 200.